ZeldaFullEditor/Gui/PaletteEditor.cs

```diff
@@ -41,79 +35,170 @@ public partial class PaletteEditor : UserControl
 
         public PaletteEditor(DungeonMain mainForm)
         {
-            InitializeComponent();
+            this.InitializeComponent();
             this.mainForm = mainForm;
 
-            CreateTempPalettes();
-            // Create temp of all palettes
-            for (int i = 0; i < 2; i++)
+            this.HudPal = new Color[Constants.HudPalettesMax][];
+            this.OverworldMainPal = new Color[Constants.OverworldMainPalettesMax][];
+            this.OverworldAuxPal = new Color[Constants.OverworldAuxPalettesMax][];
+            this.OverworldAnimatedPal = new Color[Constants.OverworldAnimatedPalettesMax][];
+            this.DungeonMainPal = new Color[Constants.DungeonsMainPalettesMax][];
+            this.GlobalSpritesPal = new Color[Constants.GlobalSpritePalettesMax][];
+            this.SpritesAux1Pal = new Color[Constants.SpritesAux1PalettesMax][];
+            this.SpritesAux2Pal = new Color[Constants.SpritesAux2PalettesMax][];
+            this.SpritesAux3Pal = new Color[Constants.SpritesAux3PalettesMax][];
+            this.ShieldsPal = new Color[Constants.ShieldsPalettesMax][];
+            this.SwordsPal = new Color[Constants.SwordsPalettesMax][];
+            this.ArmorsPal = new Color[Constants.ArmorPalettesMax][];
+
+            this.overworld_GrassPal = new Color[Constants.OverworldGrassPalettesMax];
+            this.object3D_Pal = new Color[Constants.Object3DPalettesMax][];
+            this.overworld_Maps_Pal = new Color[Constants.OverworldMiniMapPalettesMax][];
+
+            // Create temp of all palettes.
+            this.CreateTempPalettes();
+
+            this.RedoPaletteTreeNodes();
+        }
+
+        public void ResetTreeNodes()
+        {
+            this.palettesTreeView.BeginUpdate();
+
+            this.palettesTreeView.Nodes.Clear();
+
+            var treeNode1 = new TreeNode("Hud");
```

Comment:
heavily optional for making the palette names string constants but doesnt hurt to keep it consistent

Reply:
Done